src/modules/navigation/ItemNavigation.tsx

```diff
@@ -10,30 +10,42 @@ import { useMessagesTranslation } from '@/config/i18n';
 import { ROOT_ID_PATH, buildContentPagePath } from '@/config/paths';
 import { axios, hooks } from '@/config/queryClient';
 import { MAIN_MENU_ID, TREE_VIEW_ID } from '@/config/selectors';
+import { useCurrentMemberContext } from '@/contexts/CurrentMemberContext.tsx';
 import TreeView from '@/modules/navigation/tree/TreeView';
 import { isHidden } from '@/utils/item';
+import { combineUuids, shuffleAllButLastItemInArray } from '@/utils/shuffle.ts';
 
 import LoadingTree from './tree/LoadingTree';
 
 const { useItem, useDescendants, useItemsTags } = hooks;
 
 const DrawerNavigation = (): JSX.Element | null => {
   const rootId = useParams()[ROOT_ID_PATH];
+  const [searchParams] = useSearchParams();
   const navigate = useNavigate();
+  const { data: member } = useCurrentMemberContext();
   const [prevRootId, setPrevRootId] = useState(rootId);
 
   useEffect(() => {
     setPrevRootId(rootId);
   }, [rootId]);
 
+  const shuffle = Boolean(searchParams.get('shuffle') === 'true');
+
   const { t: translateMessage } = useMessagesTranslation();
 
-  const { data: descendants } = useDescendants({ id: rootId ?? '' });
+  let { data: descendants = [] } = useDescendants({ id: rootId ?? '' });
```

Comment:
Is it possible to not override this ? Instead create a derived variable from the shuffling operation.

Reply:
Do you think it's necessary? I'm happy to make another variable, but then I have to do the if/else in many places.